pymc/pytensorf.py

```diff
@@ -1234,3 +1234,68 @@ def rewrite_pregrad(graph):
     pre-grad.
     """
     return rewrite_graph(graph, include=("canonicalize", "stabilize"))
+
+
+class StringType(Type[str]):
+    def clone(self, **kwargs):
+        return type(self)()
+
+    def filter(self, x, strict=False, allow_downcast=None):
+        if isinstance(x, str):
+            return x
+        else:
+            raise TypeError("Expected a string!")
+
+    def __str__(self):
+        return "string"
+
+    @staticmethod
+    def may_share_memory(a, b):
+        return isinstance(a, str) and a is b
+
+
+stringtype = StringType()
+
+
+class StringConstant(Constant):
+    pass
+
+
+@pytensor._as_symbolic.register(str)
+def as_symbolic_string(x, **kwargs):
+    return StringConstant(stringtype, x)
+
+
+def rvs_in_graph(vars: Sequence[Variable]) -> bool:
```

Comment:
Where is this function used?

Reply:
Let's keep it there for now. Not sure yet if it's duplicated functionality or not.

Reply:
I meant to move this to the file where it's used. I'll do it now